pkg/operator/configobserver/featuregates/observe_featuregates.go

```diff
@@ -95,3 +82,47 @@ func (f *featureFlags) ObserveFeatureFlags(genericListers configobserver.Listers
 
 	return observedConfig, errs
 }
+
+func (f *featureFlags) getFeatureNames(fg *configv1.FeatureGate) ([]string, error) {
+	newConfigValue := []string{}
+	formatEnabledFunc := func(fs string) string {
+		return fmt.Sprintf("%s=true", fs)
+	}
+	formatDisabledFunc := func(fs string) string {
+		return fmt.Sprintf("%s=false", fs)
+	}
+
+	if fg.Spec.FeatureSet == configv1.CustomNoUpgrade {
+		if fg.Spec.FeatureGateSelection.CustomNoUpgrade == nil {
+			return nil, fmt.Errorf(".spec.featureGateSelection.CustomNoUpgrade is empty but the flag %s was set", configv1.CustomNoUpgrade)
+		}
+		for _, enable := range fg.Spec.FeatureGateSelection.CustomNoUpgrade.Enabled {
+			newConfigValue = append(newConfigValue, formatEnabledFunc(enable))
+		}
+		for _, disabled := range fg.Spec.FeatureGateSelection.CustomNoUpgrade.Disabled {
+			newConfigValue = append(newConfigValue, formatDisabledFunc(disabled))
+		}
+		return newConfigValue, nil
+	}
+
+	featureSet, ok := configv1.FeatureSets[fg.Spec.FeatureSet]
+	if !ok {
+		return nil, fmt.Errorf(".spec.featureSet %q not found", featureSet)
+	}
+	for _, enable := range featureSet.Enabled {
+		// only add whitelisted feature flags
+		if !f.allowAll && !f.knownFeatures.Has(enable) {
```

Comment:
I think we need this same check on the customnoupgrade or we won't be able to combine featuregates in one spot.